### ISSUE TITLE
'com.google.protobuf.gradle.ProtobufExtract' property 'destDir' is missing an input or output annotation.

### ISSUE THREAD

**Issue by alfwatt** [google/protobuf-gradle-plugin#537]
Annotation seems to be missing from:

https://github.com/google/protobuf-gradle-plugin/blob/master/src/main/groovy/com/google/protobuf/gradle/ProtobufExtract.groovy#L63

```
    dependencies {
        classpath "com.android.tools.build:gradle:7.0.3"
        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.18'
    }
```

**Comment by warreneblackwell** [comment]
* What went wrong:
Some problems were found with the configuration of task ':<PROJECT>' (type 'ProtobufExtract').
  - In plugin 'com.google.protobuf' type 'com.google.protobuf.gradle.ProtobufExtract' **property 'destDir' is missing an input or output annotation.**

    Reason: A property without annotation isn't considered during up-to-date checking.

    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.

    Please refer to https://docs.gradle.org/7.3.1/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - In plugin 'com.google.protobuf' type 'com.google.protobuf.gradle.ProtobufExtract' **property 'isTest' is missing an input or output annotation.**

    Reason: A property without annotation isn't considered during up-to-date checking.

    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.

    Please refer to https://docs.gradle.org/7.3.1/userguide/validation_problems.html#missing_annotation for more details about this problem.

**Comment by lxt-sky** [comment]
how to solve this problem?

**Comment by ejona86** [comment]
Is there a reproduction project for this? I tried to reproduce and was unable.

Our build checks this sort of thing and hasn't noticed anything. In addition, the property does have an annotation:

https://github.com/google/protobuf-gradle-plugin/blob/519849df57518f559f5cbabd2ee0298ae968a13e/src/main/groovy/com/google/protobuf/gradle/ProtobufExtract.groovy#L132-L135

Maybe the `protected` visibility confused Gradle in 7.3.1, but I suggest either upgrading to a newer Gradle version to see if it resolved the problem or providing a reproduction.

**Comment by xiaolongge904913** [comment]
I have resolved this error by updating [protobuf-gradle-plugin](https://github.com/google/protobuf-gradle-plugin) version

**Comment by ejona86** [comment]
Okay, we didn't ever reproduce, but since someone saw this go away let's call this fixed by #603 and #593, which removed isTest and replaced destDir with a public getter. Upgrade to 0.9.0 or later.

**Comment by KANGOD** [comment]
> I have resolved this error by updating [protobuf-gradle-plugin](https://github.com/google/protobuf-gradle-plugin) version

Thanks. Upgrading `protobuf-gradle-plugin` also works for me.

Verified that [v0.8.10](https://github.com/google/protobuf-gradle-plugin/blob/v0.8.10/src/main/groovy/com/google/protobuf/gradle/ProtobufExtract.groovy#L116) is lacking of the `@OutputDirectory` annotation, while [v0.8.11](https://github.com/google/protobuf-gradle-plugin/blob/v0.8.11/src/main/groovy/com/google/protobuf/gradle/ProtobufExtract.groovy#L131) fixed it.
